src/coreclr/vm/gcheaputilities.h

```diff
@@ -21,7 +60,11 @@ GVAL_DECL(GCHeapType, g_heap_type);
 // for all allocations. In order to avoid extra indirections in assembly
 // allocation helpers, the EE owns the global allocation context and the
 // GC will update it when it needs to.
-GVAL_DECL(gc_alloc_context, g_global_alloc_context);
+extern "C" ee_alloc_context g_global_ee_alloc_context;
+
+// This is a pointer into the g_global_ee_alloc_context.gc_allocation_context, the GC visible portion
```

Comment:
I do not understand why we need this extra indirection via a static variable Why can't all places that use this just take an address of `g_global_alloc_context` field?

Reply:
Do you mean take the address of g_global_ee_alloc_context.gc_allocation? That would work fine. I didn't do it that way initially just to leave the code a little closer to its current state.

Reply:
Yes, it is what I meant.